doc/proposals/scorecard-custom-tests-2.md

```diff
@@ -0,0 +1,313 @@
+ * test the custom test binary locally (out-of-cluster) to make sure the output is produced that matches v1alpha2 ScorecardTestResult format
+ * scorecard creates a scorecard DSL config file for their test image
+ * when the test binary works as expected, the developer would build and push the test image to a container registry accessible to the scorecard test environment
+ * scorecard could build a bundle image that contains the scorecard DSL config
+ NOTE:  the Basic and OLM test manifests would be added automatically to the bundle image as part of the bundle image build process.  This allows users to run the Basic and OLM tests along with their custom tests.
+ * the developer would run the scorecard passing in their operator bundle image they just built along with any other scorecard flags (e.g. selector).
+
+The case of a developer implementing a kuttl test would be as follows:
+ * developer places kuttl test assets into a kuttl test directory known to scorecard
+ * scorecard creates the scorecard DSL config file
+ * scorecard runs the tests which are based on a scorecard kuttl test image
+ * if the test results meet their goals, then scorecard could build a bundle image that contains the kuttl tests in a distributable format (e.g. bundle image)
+
+### Custom Test Packaging and Configuration
+
+In this design proposal, custom tests are container images, described in the scorecard DSL yaml file.  End users would add their scorecard DSL yaml file into their operator bundle image or pass it into scorecard using a command line flag.  Custom test images are stored in a container registry outside the control of scorecard.  The scorecard DSL command flag would take precedence over the scorecard DSL config file found in the bundle image if specified.
+
+### Scorecard Configuration
+
+The scorecard DSL yaml file would be accessible from a local disk location to scorecard or from within a bundle image. 
+
+Scorecard would assume that the scorecard DSL yaml file would be located in the bundle image at the following location:
+```
+/scorecard/config.yaml
+```
+
+The scorecard DSL format is yet to be finalized but it would need to include configuration settings for each test including labels.  Here is a sample of what the scorecard DSL configuration would look like:
+```
+tests:
+# here is an example of a pair of custom tests that an ISV might implement
+- imagename: quay.io/someuser/customtest1:v0.0.1
+  description: some ISV test
+  labels:
+    suite: custom
+    test: customtest1
+- imagename: quay.io/someuser/customtest2:v0.0.1
+  description: some ISV test
+  labels:
+    test: customtest2
+# here is an example of the scorecard basic tests
+- imagename: quay.io/operator-framework/scorecard-basictests:v0.0.1
+  description: redhat - check spec
+  labels:
+    suite: basic
+    test: basic-check-spec-test
+- imagename: quay.io/operator-framework/scorecard-basictests:v0.0.1
+  description: redhat - check status
```

Comment:
Since multiple tests can use the same image, we need to expose `entrypoint`.

Reply:
good catch, I added that in this latest commit.